File: app/src/main/java/com/dailymathtest/joy/dailymath/models/Quiz.java
package com.dailymathtest.joy.dailymath.models;

import com.google.gson.Gson;
import com.google.gson.GsonBuilder;

import java.util.Date;
import java.util.List;

public class Quiz {
    private int id;
    private double  score;
    private Date quizDate;
    private Student student;
    private List<QuizItem> quizItems;

    public int getId() {
        return id;
    }

    public double getScore() {
        return score;
    }

    public Date getQuizDate() {
        return quizDate;
    }

    public Student getStudent() {
        return student;
    }

    public List<QuizItem> getQuizItems() {
        return quizItems;
    }
    public static Quiz fromJsonStr(String jsonStr) {
        Gson gson = new GsonBuilder().setDateFormat("yyyy-MM-dd'T'HH:mm:ss").create();
        Quiz quiz = new Quiz();
        try {
            Quiz q = gson.fromJson(jsonStr, Quiz.class);
            quiz.quizItems = q.quizItems;
            quiz.student = q.student;
            quiz.score = q.score;
            quiz.id = q.id;
            quiz.quizDate = q.quizDate;
        } catch (Exception e) {
            e.printStackTrace();
            return null;
        }
        return quiz;
    }
}

File: app/src/main/java/com/dailymathtest/joy/dailymath/models/Quiz.kt
package com.dailymathtest.joy.dailymath.models

import com.google.gson.Gson
import com.google.gson.GsonBuilder

import java.util.Date

class Quiz {
    var id: Int = 0
        private set
    var score: Double = 0.toDouble()
        set
    var quizDate: Date? = null
        private set
    var student: Student? = null
        private set
    var quizItems: List<QuizItem>? = null
        private set

    companion object {
        fun fromJsonStr(jsonStr: String): Quiz? {
            val gson = GsonBuilder().setDateFormat("yyyy-MM-dd'T'HH:mm:ss").create()
            val quiz = Quiz()
            try {
                val q = gson.fromJson(jsonStr, Quiz::class.java)
                quiz.quizItems = q.quizItems
                quiz.student = q.student
                quiz.score = q.score
                quiz.id = q.id
                quiz.quizDate = q.quizDate
            } catch (e: Exception) {
                e.printStackTrace()
                return null
            }

            return quiz
        }
    }
}
